src/intrprtr.c

```diff
@@ -4183,13 +4199,17 @@ void            IntrAssComObjName (
     record = PopObj();
 
     /* assign the right hand side to the element of the record             */
-    switch (TNUM_OBJ(record)) {
-      case T_COMOBJ:
-        AssPRec( record, rnam, rhs );
-        break;
-      default:
-        ASS_REC( record, rnam, rhs );
-        break;
+    if (rhs == 0) {
+      UNB_REC( record, rnam );
+    } else {
+      switch (TNUM_OBJ(record)) {
+        case T_COMOBJ:
+          AssPRec( record, rnam, rhs );
+          break;
+        default:
+          ASS_REC( record, rnam, rhs );
+          break;
+      }
```

Comment:
Minor nitpick: This looses the optimization (?) for T_COMOBJ. But since `AssPRec` was modified to deal with rhs==0 anyway, the `if (rhs == 0)` check could be moved right next to ASS_REC.

Or, going one further, one could modify ASS_REC (andd ASS_LIST,  ASSB_LIST, ...)  to perform that check, thus making many other changes in this PR redundant. Alas, this might incur a speed penalty. Not sure how bad that would be in practice, though?